clang/include/clang/Sema/SemaConcept.h

```diff
@@ -70,7 +70,14 @@ struct alignas(ConstraintAlignment) AtomicConstraint {
     // We do not actually substitute the parameter mappings into the
     // constraint expressions, therefore the constraint expressions are
     // the originals, and comparing them will suffice.
-    if (ConstraintExpr != Other.ConstraintExpr)
+    if (AC.ConstraintExpr != Other.AC.ConstraintExpr)
+      return false;
+
+    // FIXME: As the normalization cache doesn't take
+    // ArgumentPackSubstitutionIndex into account,
+    // this won't have an effect.
+    if (AC.ArgumentPackSubstitutionIndex !=
+        Other.AC.ArgumentPackSubstitutionIndex)
```

Comment:
Github ate my comment.
This looks wrong. 

An atomic constraint should be considered equivalent to another for the purpose of subsumption if the (transformed) expression is the same, even if it is expanded from a different pack, or not expanded at all.

The only thing we should compare is
 - are they the same expression
 - do they have the same mapping

If the mapping has packs of different sizes etc _that_ should be what makes subsumption fail.

So I would expect `AtomicConstraint` not to change.

Reply:
I think in this case the constraint was expanded, the pack index should be part of the mapping.

A different pack index is notionally a different template parameter, or at least another slot where a different template argument could be located, so the different treatment wrt to non-pack parameters doesn't seem consistent to me. 

Though it's possible this doesn't make a difference in practice.
I failed to come up with an example where this would really matter, ie make something ill-formed where it's not, or vice-versa.

Reply:
Following the discussion in the reflector: http://lists.isocpp.org/core/2025/03/17677.php
There is agreement, so far, that ignoring the expansion here is wrong.

Reply:
please give me some time to think about it!

Reply:
No problem.

Since this patch fixes additional test cases, and it blocks one existing, and one soon to be posted PR, I will back-off this change and leave it for later.

Reply:
Done, this change is backed off for a future PR.